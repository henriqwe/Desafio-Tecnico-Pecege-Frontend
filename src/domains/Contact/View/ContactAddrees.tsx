import * as S from './style'
import { IContact } from '&operations/queries/fetchContacts'
import { Map } from '&components/Map/Map'

export function ContactAddrees({ contact }: { contact: IContact }) {
  return (
    <S.TabItemContainer>
      <S.InfoWrapper variant="white">
        <S.InfoTitle>Rua</S.InfoTitle>
        <S.InfoDescription>{contact.address.street || '-'}</S.InfoDescription>
      </S.InfoWrapper>
      <S.InfoWrapper variant="gray">
        <S.InfoTitle>NÃºmero</S.InfoTitle>
        <S.InfoDescription>{contact.address.suite || '-'}</S.InfoDescription>
      </S.InfoWrapper>
      <S.InfoWrapper variant="white">
        <S.InfoTitle>Cidade</S.InfoTitle>
        <S.InfoDescription>{contact.address.city || '-'}</S.InfoDescription>
      </S.InfoWrapper>
      <S.InfoWrapper variant="gray">
        <S.InfoTitle>CEP</S.InfoTitle>
        <S.InfoDescription>{contact.address.zipcode || '-'}</S.InfoDescription>
      </S.InfoWrapper>
      {contact.address.geo.lat && contact.address.geo.lng && (
        <S.MapContainer>
          <Map contacts={[contact]} showInfoWindow={false} />
        </S.MapContainer>
      )}
    </S.TabItemContainer>
  )
}
