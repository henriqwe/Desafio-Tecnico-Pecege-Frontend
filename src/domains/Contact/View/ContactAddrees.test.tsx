import { render, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'
import { TestProviders } from '&utils/tests/TestProviders'
import { ContactAddrees } from './ContactAddrees'

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useParams: () => ({
    id: 1
  })
}))
const contact = {
  id: 3,
  name: 'Clementine Bauch',
  username: 'Samantha',
  email: 'Nathan@yesenia.net',
  address: {
    street: 'Douglas Extension',
    suite: 'Suite 847',
    city: 'McKenziehaven',
    zipcode: '59590-4157',
    geo: {
      lat: '-68.6102',
      lng: '-47.0653'
    }
  },
  phone: '1-463-123-4447',
  website: 'ramiro.info',
  company: {
    name: 'Romaguera-Jacobson',
    catchPhrase: 'Face to face bifurcated interface',
    bs: 'e-enable strategic applications'
  }
}

describe('ContactAddrees', () => {
  it('should render the contact list correctly', async () => {
    const { container, findByTestId } = render(
      <TestProviders>
        <ContactAddrees contact={contact} />
      </TestProviders>
    )

    await waitFor(async () => findByTestId('tab-container'))

    const street = (await findByTestId('info-street')) as HTMLParagraphElement
    const suite = (await findByTestId('info-suite')) as HTMLParagraphElement
    const city = (await findByTestId('info-city')) as HTMLParagraphElement
    const zipcode = (await findByTestId('info-zipcode')) as HTMLParagraphElement

    expect(street.innerHTML).toEqual(contact.address.street)
    expect(suite.innerHTML).toEqual(contact.address.suite)
    expect(city.innerHTML).toEqual(contact.address.city)
    expect(zipcode.innerHTML).toEqual(contact.address.zipcode)

    expect(container).toMatchSnapshot()
  })
})
