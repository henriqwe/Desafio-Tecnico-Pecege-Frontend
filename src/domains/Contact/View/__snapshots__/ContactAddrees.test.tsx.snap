// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactAddrees should render the contact list correctly 1`] = `
<div>
  <section
    class="sc-csmVar kBNagx"
    data-testid="tab-container"
  >
    <div
      style="width: 100%; display: flex; opacity: 0.18119594527698268; transform: translateX(-244.11672780889091px) translateZ(0);"
    >
      <div
        class="sc-bjEwCx cdROzo"
      >
        <div
          class="sc-eLSjS gLXBOl"
          variant="white"
        >
          <p
            class="sc-kDnyCx iVwbOt"
          >
            Rua
          </p>
          <p
            class="sc-ipMvLY fVVflE"
            data-testid="info-street"
          >
            Douglas Extension
          </p>
        </div>
        <div
          class="sc-eLSjS cJvzVk"
          variant="gray"
        >
          <p
            class="sc-kDnyCx iVwbOt"
          >
            NÃºmero
          </p>
          <p
            class="sc-ipMvLY fVVflE"
            data-testid="info-suite"
          >
            Suite 847
          </p>
        </div>
        <div
          class="sc-eLSjS gLXBOl"
          variant="white"
        >
          <p
            class="sc-kDnyCx iVwbOt"
          >
            Cidade
          </p>
          <p
            class="sc-ipMvLY fVVflE"
            data-testid="info-city"
          >
            McKenziehaven
          </p>
        </div>
        <div
          class="sc-eLSjS cJvzVk"
          variant="gray"
        >
          <p
            class="sc-kDnyCx iVwbOt"
          >
            CEP
          </p>
          <p
            class="sc-ipMvLY fVVflE"
            data-testid="info-zipcode"
          >
            59590-4157
          </p>
        </div>
      </div>
    </div>
    <div
      style="width: 100%; height: 100%; display: flex; background-color: red; opacity: 0.1710638766022734; transform: translateX(248.68083701931803px) translateZ(0);"
    >
      <div
        class="sc-fnOeiS bqivNc"
      />
    </div>
  </section>
</div>
`;
